tests/EcbCurrencyRateProviderTest.php

```diff
@@ -87,7 +88,8 @@ function (ICurrency $currency, ICurrencyRateProvider $provider, \DateTime $date,
 
         $this->currencyRateProvider = new EcbCurrencyRateProvider(
             $currencyRateManager,
-            $currencyManager
+            $currencyManager,
+            new XMLLoader()
```

Comment:
You did excellent job about XMLLoader, this is actually I wanted to ask you to do.

Because it will improve tests.
And from now during the tests it's not necessary to request real providers for data.
You should mock XMLLoader::load method and return general XML and error XML to check some of your exceptions!